### DISCUSSION TITLE
Overall performance leaderboards

### DISCUSSION THREAD

**Discussion by powwu** [monkeytypegame/monkeytype#842] [Ideas] [answered]
## What?
I think there should be some sort of a completely global, everyone is ranked type of leaderboard system.

## Why?
Adding a simple number (a "rank", as it would possibly be called) would allow for healthy competition within the Monkeytype community. It's not satisfying to be improving yourself, just to not qualify in races. Everyone should receive a rank, based on their overall speed, allowing for you to compare with friends.

## How?
There are quite a few ways one could go about this.

The way I see as the most appealing would be to have a weighting system, similar to [osu!](http://osu.ppy.sh)'s performance point system. It would work in this context as follows:

You have a total performance number that reflects your rank. The person with the highest performance number gets #1, the second #2, so on and so forth. The more tests you submit, the more your score raises. To prevent people from submitting low effort tests, a weighting system would be applied.

Your highest WPM would be added to your performance score as-is
95% of your second highest WPM would be added to your performance score
90% of your third highest WPM would be added to your performance score

so on, and so forth, until scores that don't break WPM boundaries won't be worth much.

In this case, there would be two leaderboards. One for 15 seconds, and one for 60 seconds.

## Other ways this could be done
Since multiplayer is in development, I think a satisfactory idea would be having a score that rises depending on how many people you beat in multiplayer. This score would reset every week, as a "season", similar to various games (such as WoW)'s PVP system.

Thank you for hearing me out,
Powwu

**Comment by Miodec** [comment, marked as answer]
It would be nice, but doing this at the moment isn't really realistic due to the billing situation of the database. A system like this would require a lot of extra reads and data storage.

Something similar is planned in a form of ranked 1v1 mode with a typical elo system, but thats in the future.

**Reply by powwu** [reply]
Perfect, a ranked 1v1 mode sounds fun. Looking forward to it!

**Reply by AvidItalianCoffeeDrinker456** [reply]
> It would be nice, but doing this at the moment isn't really realistic due to the billing situation of the database. A system like this would require a lot of extra reads and data storage.
> 
> Something similar is planned in a form of ranked 1v1 mode with a typical elo system, but thats in the future.

a few extra bytes on everyone who uses the site cant be that hurtful

**Comment by dominikgorczyca** [comment]
It would be nice if there was leaderboard for 30 seconds too.
